src/internal.c

```diff
@@ -10899,6 +10904,8 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
                         if (args->lastErr == 0) {
                             args->lastErr = ret; /* save error from last time */
                             ret = 0; /* reset error */
+                        } else if (ssl->options.verifyNone) {
+                            ret = 0; /* reset error */
```

Comment:
What does this have to do with CRL?  And why all ALL errors cleared here?

Reply:
If more than one non-leaf cert is being processed in this loop, then only the first pass gets reset  (because `args->lastErr == 0` is still true). But the next time through, the error in ret was not being reset even though verify none was specified.

Reply:
The title of the PR is about CRLs.  This part of the code doesn't appear to have anything to do with CRLs.  That is, non CRLs will have ALL errors reset here.  That's a pretty big change to make and not all what the title or comment says.

Reply:
I removed this reset of the return value. In fact, `ret` will not be assigned a new value after the verify callback if `NO_VERIFY` is specified. Thanks for your feedback!